infra/modules/common_environment/main.tf

```diff
@@ -0,0 +1,123 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.111.0"
```

Comment:
Can we move to 4.x?

Reply:
Done, but for some restriction it install the 3.116

Reply:
Ah probably for `terraform-azurerm-v3 repository`.. we may plan to remove that dependency in next prs